smart-py/objkt_swap_tests.py

```diff
@@ -0,0 +1,852 @@
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    swap = OBJKTSwap(
+        objkt.address,
+        hdao.address,
+        manager.address,
+        metadata,
+        curator.address
+    )
+
+    scenario += swap
+
+    # default variables that get set in the contract constructor
+    # TODO find out if these are the intended initial values
+    scenario.verify(swap.data.swap_id == 0)
+    scenario.verify(swap.data.objkt_id == 152)
+    scenario.verify(swap.data.genesis == sp.timestamp(0))
+    scenario.verify(swap.data.locked == False)
+
+    # addresses are as expected
+    scenario.verify(swap.data.manager == manager.address)
+    scenario.verify(swap.data.hdao == hdao.address)
+    scenario.verify(swap.data.curate == curator.address)
+    scenario.verify(swap.data.objkt == objkt.address)
+
+    # default values
+    # TODO cannot be tested due to locally declared variables inside
+    # scenario.verify(swap.royalties == 0)
+    # scenario.verify(swap.fee == 0)
+    # scenario.verify(swap.amount == 0)
+
+    # metadata
+    # TODO how does this work? am i passing in the metadata correctly?
+    # scenario.verify(swap.data.metadata.name == 'something')
+
+    # TODO how to test these members?
+    # print(vars(swap.data.swaps))
+    # print(vars(swap.data.royalties))
+
+@sp.add_test("Test genesis entrypoint")
+def test():
+    # init test and create html output
+    scenario = sp.test_scenario()
+    scenario.h1("Genesis Test")
+
+    # init test values
+    seller = sp.test_account("seller")
+    manager = sp.test_account("manager")
+    curator = sp.test_account("curator")
+    objkt = sp.test_account("objkt123")
+    hdao = sp.test_account("hdao")
+
+    creator = sp.test_account("creator")
+    metadata = sp.record(
+        name = "test",
+        description = "test",
+        tags = [
+            'test'
+        ],
+        symbol = 'OBJKT',
+        artifactUri = "ipfs://test",
+        displayUri = "ipfs://test",
+        thumbnailUri = "ipfs://test",
+        creators = [
+            creator.address
+        ],
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    swap = OBJKTSwap(
+        objkt.address,
+        hdao.address,
+        manager.address,
+        metadata,
+        curator.address
+    )
+
+    scenario += swap
+
+    # illegal attempts
+    scenario += swap.genesis().run(sender = seller, valid = False)
+    scenario += swap.genesis().run(sender = hdao, valid = False)
+    scenario += swap.genesis().run(sender = curator, valid = False)
+    scenario += swap.genesis().run(sender = objkt, valid = False)
+
+    # valid attempt
```

Comment:
Should there also be tests to verify the genesis() call set the swap data correctly?

Reply:
added more tests